lua/sp.c

```diff
@@ -3238,6 +3242,21 @@ static int join_threads(SP sp)
 static int dbstmt_free(Lua L)
 {
     dbstmt_t *stmt = lua_touserdata(L, -1);
+    Vdbe *pVdbe = (Vdbe*)stmt->stmt;
+
+    /* Compute and add fingerprint for this statement in case it wasn't
```

Comment:
Maybe this could just call lua_end_step?

Reply:
Great idea Joe! Updated the patch.

Reply:
Cool... One minor thing: The (pVdbe->luaStartTime != 0) check in this function seems superfluous since it's already done via lua_end_step().

Reply:
Done.